packages/components/src/text/hook.ts

```diff
@@ -64,9 +68,7 @@ export default function useText( props ) {
 
 		content = createHighlighterText( {
 			autoEscape: highlightEscape,
-			// Disable reason: We need to disable this otherwise it erases the cast
-			// eslint-disable-next-line object-shorthand
-			children: /** @type {string} */ ( children ),
+			children: children as string,
```

Comment:
The type cast shouldn't be needed, given the type guard a few lines above

```suggestion
			children,
```

Reply:
Done.